src/app/components/source/UserSecurity.js

```diff
@@ -0,0 +1,522 @@
+    };
+    let { sendSuccessfulLogin, sendFailedLogin } = this.state;
+    if (type === 'successfulLogin') {
+      sendSuccessfulLogin = inputChecked;
+    } else {
+      sendFailedLogin = inputChecked;
+    }
+    this.setState({
+      sendSuccessfulLogin,
+      sendFailedLogin,
+    });
+    Relay.Store.commitUpdate(
+      new SetUserSecuritySettingsMutation({
+        id,
+        sendSuccessfulLogin,
+        sendFailedLogin,
+      }),
+      { onSuccess, onFailure },
+    );
+  }
+
+  handleGenerateBackupCodes() {
+    const { dbid } = this.props.user;
+
+    const onFailure = () => {};
+
+    const onSuccess = (response) => {
+      const { generateTwoFactorBackupCodes: { codes } } = response;
+      this.setState({ backupCodes: codes });
+    };
+
+    Relay.Store.commitUpdate(
+      new GenerateTwoFactorBackupCodesMutation({
+        id: dbid,
+      }),
+      { onSuccess, onFailure },
+    );
+  }
+
+  renderMessage(item) {
+    switch (item) {
+    case 'passwordInput':
+      return this.props.intl.formatMessage(messages.passwordInput);
+    case 'passwordError':
+      return this.props.intl.formatMessage(messages.passwordError);
+    case 'verifyInput':
+      return this.props.intl.formatMessage(messages.verifyInput);
+    case 'verifyError':
+      return this.props.intl.formatMessage(messages.verifyError);
+    default:
+      return null;
+    }
+  }
+
+  render() {
+    const { user } = this.props;
+    const { can_enable_otp, qrcode_svg } = user.two_factor;
+    const currentUser = this.getCurrentUser();
+
+    if (!currentUser || !user || currentUser.dbid !== user.dbid) {
+      return null;
+    }
+
+    const style = {
+      margin: `${units(2)} 0`,
+    };
+
+    const cardStyle = {
+      margin: `${units(2)} 0`,
+    };
+
+    const cardTextStyle = {
+      display: 'flex',
+      alignItems: 'center',
+    };
+
+    const cardTextAuthStyle = {
+      display: 'table-row',
+      alignItems: 'center',
+    };
+
+    const subTitleStyle = {
+      margin: '5px 0px',
+      textTransform: 'uppercase',
+    };
+
+    const divBackupStyle = {
+      lineHeight: '25px',
+      fontWeight: 'bold',
+      fontSize: '12px',
+      backgroundColor: '#dfdfe6',
+      margin: '5px',
+      textAlign: 'center',
+    };
+
+    const appsUrls = {
+      apple: 'https://apps.apple.com/us/app/google-authenticator/id388497605',
+      play: 'https://play.google.com/store/apps/details?id=com.google.android.apps.authenticator2',
+      appleImage: '/images/two_factor/apple.png',
+      playImage: '/images/two_factor/play.png',
+    };
+
+    // TODO: Read loginTrail from config
+    const loginTrial = 4;
+
+    return (
+      <div id="user__security">
+        <h2 style={style}>
+          <FormattedMessage id="userSecurity.notification" defaultMessage="Notification" />
+        </h2>
+        <Card style={cardStyle}>
+          <CardText style={cardTextStyle}>
+            <span style={{ minWidth: '500px', padding: '0px' }}>
```

Comment:
We don't set sizes in pixels, instead we use `units()` for a 8px grid normalization. Please use `units()` instead of specifying sizes in pixels directly.

Reply:
Done